src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -105,6 +105,7 @@ public void GenSspiClientContext(SspiClientContextStatus sspiClientContextStatus
 
             ContextFlagsPal requestedContextFlags = ContextFlagsPal.Connection
                 | ContextFlagsPal.Confidentiality
+                | ContextFlagsPal.Delegate
                 | ContextFlagsPal.MutualAuth;
```

Comment:
We had a look at why the Windows version works without this change, and suspect that it automatically gets a ticket with the Delegate flag already set?

These flags do not seem to get modified between here and where they get passed into the low level GSSAPI library, so we couldn't find a better place than here to add the Delegate flag..